opm/simulators/utils/readDeck.cpp

```diff
@@ -225,31 +227,63 @@ void readDeck(int rank, std::string& deckFilename, std::unique_ptr<Opm::Deck>& d
                 eclipseState = std::make_unique<Opm::EclipseState>(*deck);
 #endif
             }
-            /*
-              For the time being initializing wells and groups from the
-              restart file is not possible, but work is underways and it is
-              included here as a switch.
-            */
+
             const auto& init_config = eclipseState->getInitConfig();
-            if (init_config.restartRequested() && initFromRestart) {
-                int report_step = init_config.getRestartStep();
-                const auto& rst_filename = eclipseState->getIOConfig().getRestartFileName( init_config.getRestartRootName(), report_step, false );
-                Opm::EclIO::ERst rst_file(rst_filename);
-                const auto& rst_state = Opm::RestartIO::RstState::load(rst_file, report_step);
-                if (!schedule)
-                    schedule = std::make_unique<Opm::Schedule>(*deck, *eclipseState, *parseContext, *errorGuard, python, outputInterval, &rst_state);
+            if (init_config.restartRequested()) {
+                // Analytic aquifers must always be loaded from restart file
+                // in restarted runs because the corresponding keywords do
+                // not exist in the input deck in this case.  In other
+                // words, there's no way to check if there really are
+                // analytic aquifers in the run until we attempt to read the
+                // specifications from the restart file.  If the loader
+                // determines that there are no analytic aquifers, then
+                // 'getAnalyticAquifers()' does nothing.
+                const int report_step = init_config.getRestartStep();
+                const auto rst_filename = eclipseState->getIOConfig().getRestartFileName( init_config.getRestartRootName(), report_step, false );
+                auto rst_file = std::make_shared<Opm::EclIO::ERst>(rst_filename);
+
```

Comment:
I have of course not looked in detail on all of this - but I would really like some structural change:

1. I would prefer if the `RstState::load()` was brought out to the current scope and the `RstState` "class" gets the basic data-transfer support for Aquifers.
2. When it comes to the Aquifer configuration information the `EclipseState` class is the "main manager" - so rather than setting aquifer configuration on the `EclipseState` with:
```
aquifers.getAnalyticAquifers( eclipseState->mutableAquifers() );
```
I would prefer if this was implemented with:
```
EclipseState::load_rst_aquifers( RstState::aquifers );
```